### DISCUSSION TITLE
Handling page-breaks

### DISCUSSION THREAD

**Discussion by zusamarehan** [spatie/browsershot#582] [General]
Hi Team, 

I have been using this package from a day or two for generating my Invoices, because of the reason I was able to use TailwindCSS. 

One of the problems which I am facing right now is page breaks. What if the table spans across multiple pages, or what if we want to manually break the page for some reason? 

I would highly aprreciate for some positive reply. Thanks!

**Comment by chrisbalicki** [comment]
This is the only way I have managed to get page breaks working OK, its still not great. 

First create a page break...
`  <style>

    @media print {
      .pagebreak { page-break-before: always; } /* page-break-after works, as well */
    }

    @media print {
      body {-webkit-print-color-adjust: exact;} /* I am using this to make Tailwind colours i.e bg-gray-300 etc.. to also print in the PDF */
    }

  </style>`


then just add the following page break where you want it 

`<div class="pagebreak"> </div>`

I was facing an issue with the break just splitting the row across two pages, I got around this by breaking inside, so if it breaks on an item it will push to the next page... 

Add this to your tr tag

`style="
        -webkit-column-break-inside: avoid;
        page-break-inside: avoid;
        break-inside: avoid;" `

**Comment by betsyecastro** [comment]
Has anyone had luck? @zusamarehan 

Did the solution provided by @chrisbalicki work for you?

Thanks

**Comment by codepotato** [comment]
The modern way to tackle this is by using the following code: 

```
.pagebreak {
  break-after: page
}
```

**Reply by venuscreats** [reply]
this is worked for me with Laravel project , Browsershot PDF download.

**Reply by tominal** [reply]
Love it! Thanks for posting this.
